src/services/checkout/CheckoutService.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+namespace App\services\checkout;
+
+use Closure;
+use Dotenv\Dotenv;
+use Xendit\Xendit;
+
+class CheckoutService {
+
+    public function createInvoice($args) {
+        // $data = json_decode(file_get_contents('php://input'), true);
+        // return json_encode($data);
+        $env = Dotenv::createUnsafeImmutable(dirname(BASEPATH));
+        $env->load();
+        $apiKey = $this->env('API_KEY');
+        // var_dump($apiKey);
+
+        $date = new \DateTime();
+        $redirectUrl = '';
+        $defParams = [
+            'external_id' => 'native8-checkout-demo-' . $date->getTimestamp(),
+            'payer_email' => 'invoice+demo@xendit.co', 
+            'description' => 'Vanilla PHP Checkout Demo', 
+            'failure_redirect_url' => $redirectUrl, 
+            'success_redirect_url' => $redirectUrl
+        ];
+
+        $defParams['failure_redirect_url'] = $args['redirect_url'];
+        $defParams['success_redirect_url'] = $args['redirect_url'];
+        $post = [];
+
+        foreach ($args as $k => $v) {
+            $post[$k] = $v;
+        }
+        
+        $params = array_merge($defParams, $post);
+
+        // return json_encode($params);
+
+        Xendit::setApiKey($apiKey);
+        header('Conten-Type: application/json');
+
+        $createdInvoice = \Xendit\Invoice::create($params);
```

Comment:
Please handle Try and Catch here. If the API call is failing, should return error message to front end.

Attached:
![image](https://user-images.githubusercontent.com/90175540/136322478-663be568-734b-40b9-8052-5c09e07098b7.png)

Something like this:
```
$resp = [];
try{
    $createdInvoice = \Xendit\Invoice::create($params);
    $resp =  $createdInvoice;
}catch(ApiException $e){
    $resp['message'] = $e->getMessage();
}

return json_encode($resp);
```
Just a reference, you may or may not follow. You can do it with your style.